### DISCUSSION TITLE
GitHub Action: POSIX "command" command unavailable in ubuntu-latest?

### DISCUSSION THREAD

**Discussion by pmonks** [community/community#26788] [Actions]
<p>I have a workflow action implemented in Clojure (i.e. running on the JVM) that, as part of its processing, forks a <code>command -v [something]</code> command to check whether a given binary executable is available or not.  However the result is ENOENT (errno 2 - file not found).  The <code>runs-on</code> is <code>ubuntu-latest</code>.</p>
<p>You can see an example of this <a href="https://github.com/pmonks/tools-convenience/runs/4003514241?check_suite_focus=true#step:5:349" rel="noopener nofollow ugc">in this build output</a>.</p>
<p>I also tried fully qualifying it with <code>/usr/bin/command</code>, but that failed with <a href="https://github.com/pmonks/tools-convenience/runs/4003619369?check_suite_focus=true#step:5:349" rel="noopener nofollow ugc">much the same error</a>.</p>
<p>I have a hard time believing this is a limitation of the GitHub Action ubuntu environment, given that <code>command</code> is a basic element of POSIX, but I’m stumped as to what I’ve done wrong to make it unavailable.  Any ideas or suggestions?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p><code>command</code> is a built-in function of the POSIX shell, so there is no binary. You can call a shell to run your command:</p>
<pre><code class="lang-auto">sh -c "command -v [something]"
</code></pre>
<p>Warning: If the “something” is user-provided or from another untrusted source that would be vulnerable to shell injection. :warning:</p>

**Comment by pmonks** [comment]
<p>Not necessarily.  There’s a <code>/usr/bin/command</code> binary on many POSIX systems (most BSD derivatives, for example, as well as Solaris).  Shame Linux doesn’t follow suit…  Anyhoo, thanks for the tip.</p>
<p>And yes, <code>[something]</code> is a hardcoded string literal that is not constructed in whole or part from user input or data from the environment.</p>
